app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditViewModel.kt

```diff
@@ -372,22 +397,46 @@ class VaultAddEditViewModel @Inject constructor(
     private fun registerFido2Credential(
         request: Fido2CredentialRequest,
         content: VaultAddEditState.ViewState.Content,
+    ) {
+        val createOptions = fido2CredentialManager
+            .getPasskeyCreateOptionsOrNull(request.requestJson)
+            ?: run {
+                showFido2ErrorDialog()
+                return
+            }
+
+        when (createOptions.authenticatorSelection.userVerification) {
+            UserVerificationRequirement.DISCOURAGED -> {
+                registerFido2CredentialToCipher(request, content.toCipherView())
+            }
+
+            UserVerificationRequirement.PREFERRED -> {
+                sendEvent(VaultAddEditEvent.Fido2UserVerification(isRequired = false))
+            }
+
+            UserVerificationRequirement.REQUIRED,
+            null,
```

Comment:
Is this the same thing where the `null` case should be separated?

Reply:
Indeed. Thank you.